pandas/core/frame.py

```diff
@@ -8548,6 +8462,140 @@ def isin(self, values) -> "DataFrame":
 ops.add_special_arithmetic_methods(DataFrame)
 
 
+@functools.singledispatch
```

Comment:
if we are moving this i think moving this into
pandas/frame/construction.py would be fine

Reply:
Sounds good.